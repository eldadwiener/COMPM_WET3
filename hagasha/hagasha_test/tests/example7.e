COPYI I1 0
COPYI I2 0
COPYI I3 0
COPYI I4 0
COPYI I5 0
COPYI I6 0
COPYI I7 0
JLINK 42
HALT
COPYI I3 1
STORI I3 I1 -4
COPYI I2 I1
RETRN
COPYI I2 I1
RETRN
ADD2I I2 I2 4
STORI I0 I2 0
STORI I1 I2 4
STORI I2 I2 8
CITOF F0 I2
ADD2I I2 I2 16
COPYI I1 I2
JLINK 10
SUBTI I2 I2 16
LOADI I0 I2 0
LOADI I1 I2 4
LOADI I2 I2 8
LOADI I3 I2 12
STORI I3 I1 0
LOADI I4 I1 0
LOADI I5 I1 -8
LOADI I6 0 I5
ADD2I I4 I4 I6
STORI I4 I1 0
LOADI I7 I1 0
STORI I7 I1 -4
COPYI I2 I1
RETRN
COPYI I2 I1
RETRN
HALT
PRNTC 10
PRNTC 84
PRNTC 101
PRNTC 115
PRNTC 116
PRNTC 58
PRNTC 32
PRNTC 112
PRNTC 97
PRNTC 115
PRNTC 115
PRNTC 105
PRNTC 110
PRNTC 103
PRNTC 32
PRNTC 112
PRNTC 111
PRNTC 105
PRNTC 110
PRNTC 116
PRNTC 101
PRNTC 114
PRNTC 32
PRNTC 116
PRNTC 111
PRNTC 32
PRNTC 102
PRNTC 117
PRNTC 110
PRNTC 99
PRNTC 116
PRNTC 105
PRNTC 111
PRNTC 110
PRNTC 32
PRNTC 105
PRNTC 110
PRNTC 32
PRNTC 100
PRNTC 105
PRNTC 102
PRNTC 102
PRNTC 101
PRNTC 114
PRNTC 101
PRNTC 110
PRNTC 116
PRNTC 32
PRNTC 102
PRNTC 105
PRNTC 108
PRNTC 101
PRNTC 10
ADD2I I2 I2 4
ADD2I I2 I2 8
ADD2I I2 I2 4
COPYI I3 2
STORI I3 I1 12
LOADI I4 I1 12
ADD2I I5 I1 12
STORI I5 I1 0
LOADI I6 I1 0
STORI I0 I2 0
STORI I1 I2 4
STORI I2 I2 8
STORI I3 I2 12
STORI I4 I2 16
STORI I5 I2 20
STORI I6 I2 24
CITOF F0 I2
STORI I6 I2 28
ADD2I I2 I2 36
COPYI I1 I2
JLINK 16
SUBTI I2 I2 36
LOADI I0 I2 0
LOADI I1 I2 4
LOADI I2 I2 8
LOADI I3 I2 12
LOADI I4 I2 16
LOADI I5 I2 20
LOADI I6 I2 24
LOADI I7 I2 32
STORI I7 I1 8
PRNTC 115
PRNTC 104
PRNTC 111
PRNTC 117
PRNTC 108
PRNTC 100
PRNTC 32
PRNTC 98
PRNTC 101
PRNTC 32
PRNTC 51
PRNTC 58
PRNTC 32
LOADI I8 I1 8
PRNTI I8
PRNTC 10
LOADI I9 I1 0
STORI I0 I2 0
STORI I1 I2 4
STORI I2 I2 8
STORI I3 I2 12
STORI I4 I2 16
STORI I5 I2 20
STORI I6 I2 24
STORI I7 I2 28
STORI I8 I2 32
STORI I9 I2 36
CITOF F0 I2
STORI I9 I2 40
ADD2I I2 I2 48
COPYI I1 I2
JLINK 16
SUBTI I2 I2 48
LOADI I0 I2 0
LOADI I1 I2 4
LOADI I2 I2 8
LOADI I3 I2 12
LOADI I4 I2 16
LOADI I5 I2 20
LOADI I6 I2 24
LOADI I7 I2 28
LOADI I8 I2 32
LOADI I9 I2 36
LOADI I10 I2 44
LOADI I11 I1 12
LOADI I12 I1 12
SGRTI I13 I10 I12
BNEQZ I13 175
UJUMP 177
COPYI I13 20
UJUMP 180
COPYI I14 10
STORI I14 I1 4
UJUMP 181
STORI I13 I1 4
LOADI I15 I1 0
STORI I0 I2 0
STORI I1 I2 4
STORI I2 I2 8
STORI I3 I2 12
STORI I4 I2 16
STORI I5 I2 20
STORI I6 I2 24
STORI I7 I2 28
STORI I8 I2 32
STORI I9 I2 36
STORI I10 I2 40
STORI I11 I2 44
STORI I12 I2 48
STORI I13 I2 52
STORI I14 I2 56
STORI I15 I2 60
CITOF F0 I2
STORI I15 I2 64
ADD2I I2 I2 72
COPYI I1 I2
JLINK 16
SUBTI I2 I2 72
LOADI I0 I2 0
LOADI I1 I2 4
LOADI I2 I2 8
LOADI I3 I2 12
LOADI I4 I2 16
LOADI I5 I2 20
LOADI I6 I2 24
LOADI I7 I2 28
LOADI I8 I2 32
LOADI I9 I2 36
LOADI I10 I2 40
LOADI I11 I2 44
LOADI I12 I2 48
LOADI I13 I2 52
LOADI I14 I2 56
LOADI I15 I2 60
LOADI I16 I2 68
LOADI I17 I1 12
LOADI I18 I1 12
SLETI I19 I16 I18
BNEQZ I19 226
UJUMP 228
COPYI I19 20
UJUMP 231
COPYI I20 10
STORI I20 I1 8
UJUMP 232
STORI I19 I1 8
PRNTC 115
PRNTC 104
PRNTC 111
PRNTC 117
PRNTC 108
PRNTC 100
PRNTC 32
PRNTC 98
PRNTC 101
PRNTC 32
PRNTC 50
PRNTC 48
PRNTC 58
PRNTC 32
LOADI I21 I1 4
PRNTI I21
PRNTC 10
PRNTC 115
PRNTC 104
PRNTC 111
PRNTC 117
PRNTC 108
PRNTC 100
PRNTC 32
PRNTC 98
PRNTC 101
PRNTC 32
PRNTC 49
PRNTC 48
PRNTC 58
PRNTC 32
LOADI I22 I1 8
PRNTI I22
PRNTC 10
COPYI I2 I1
RETRN
HALT
