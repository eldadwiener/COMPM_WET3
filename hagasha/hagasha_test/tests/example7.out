
Test: passing pointer to function in different file
should be 3: 3 
should be 20: 20 
should be 10: 10 
Reached Halt.
